server/routes/private.js

```diff
@@ -136,11 +137,26 @@ router.get('/appointments', async (req, res) => {
 });
 
 
-//Returns all possible timeslots
-// Returns all the appointments/meetings for a given student or worker.
-// Note that appointments/meetings are synonymous, but only appointments will be used in the backend to maintain consistency.
+
+// This method returns the timeslot records stored in the timeslot table 
+//Note the parameters, as this method, if given specific start time and specific end time to return, wil only return timeslots within the start and end time parameters 
+//If one or both ofthe parameters is null or not vlaid, the method will return all timeslots in the timeslot table 
 router.get('/possible-timeslots', async (req, res) => {
-timeslots = await workerTimeslotHandler.getPossibleTimeslots();
+    const paramSchema = Joi.object({
+        specificStartTime: Joi.date().iso(),
+        specificEndTime: Joi.date().iso().greater(Joi.ref('startTime'))
+    });
+
+    const query = req.query ? req.query : {};
+
+    const specificStartTime = query.specificStartTime ? query.specificStartTime : null;
+    const specificEndTime = query.specificEndTime ? query.specificEndTime : null;
+
+    //const { error } = paramSchema.validate({specificStartTime, specificEndTime });
```

Comment:
Hi Peter,
Is it possible to explain why these lines were commented out? Based on the current format, an error would not be thrown if there were violations against the specificStartTime and specificEndTime, e.g., if they were not of ISO date format, for instance.

Reply:
Based on my endpoint testing, if i leave lines 155 and 157, then when null parameters are passed, the method throws an error rather than returning all possible timeslots. I will test by uncommenting only line 155 and seeing what happens.

Reply:
Confirmed that both lines need to be commented out for null parameters to be successfully passed.

Reply:
Hi Peter,
The reason that you are encountering that specific error is because of line 152 and line 153. When the parameters that are passed in are not defined, you set them to null. However, Joi only accepts undefined OR values that follow the specific format (e.g., iso date strings in your method). My recommendation would be to EITHER remove the latter part of lines 152 and 153 OR allow the values to be undefined in your Joi parameter scheme on lines 146 and 147.

Reply:
Comment relevant to a development process that has been shelved.